test/unit/ui.js

```diff
@@ -423,5 +427,123 @@ describe('ui class', () => {
         });
       });
     });
+
+    describe('modal methods', () => {
+      describe('createModal()', () => {
+        it('creates a new modal if one does not already exist', () => {
+          ui.createModal({options: {}});
+          assert.equal(1, ui.components.modal.length);
+          assert.instanceOf(ui.components.modal[0], Modal);
+        });
+
+        it('does not create a modal if it already exists', () => {
+          ui.components.modal = [{options: {}}];
+          ui.createModal({options: {}});
+          assert.equal(1, ui.components.modal.length);
+        });
+
+        it('returns the newly created modal', () => {
```

Comment:
As per our IRL convo, let's include these test conditions to the previous tests to maintain context.